proto/protocolapis/astria/protocol/transaction/v1/action.proto

```diff
@@ -233,19 +232,33 @@ message FeeChange {
     astria.protocol.fees.v1.SudoAddressChangeFeeComponents sudo_address_change = 12;
     astria.protocol.fees.v1.TransferFeeComponents transfer = 13;
     astria.protocol.fees.v1.ValidatorUpdateFeeComponents validator_update = 14;
-    astria.protocol.fees.v1.AddCurrencyPairsFeeComponents add_currency_pairs = 15;
-    astria.protocol.fees.v1.RemoveCurrencyPairsFeeComponents remove_currency_pairs = 16;
+    astria.protocol.fees.v1.PriceFeedFeeComponents price_feed = 15;
   }
 }
 
 message IbcSudoChange {
   astria.primitive.v1.Address new_address = 1;
 }
 
-message AddCurrencyPairs {
+// A transaction that modifies the price feed oracle settings.
+message PriceFeed {
+  oneof value {
+    CurrencyPairsChange oracle = 1;
+  }
+}
+
+// Add or remove currency pairs to/from the price feed oracle.
+message CurrencyPairsChange {
+  oneof value {
+    CurrencyPairsToAdd addition = 1;
+    CurrencyPairsToRemove removal = 2;
+  }
+}
+
+message CurrencyPairsToAdd {
   repeated connect.types.v2.CurrencyPair pairs = 1;
 }
 
-message RemoveCurrencyPairs {
+message CurrencyPairsToRemove {
   repeated connect.types.v2.CurrencyPair pairs = 1;
 }
```

Comment:
nit but since these two are the same type, can we use just one type `CurrencyPairs` and determine addition or removal based on enum variant?

Reply:
Good point.  Done in [1180e60](https://github.com/astriaorg/astria/pull/1900/commits/1180e60eafa5da1188e7e71d5b47c009acbbe585).